src/shims/env.rs

```diff
@@ -111,4 +112,33 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             Ok(-1)
         }
     }
+
+    fn getcwd(
+        &mut self,
+        buf_op: OpTy<'tcx, Tag>,
+        size_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, Scalar<Tag>> {
+        let this = self.eval_context_mut();
+
+        if this.machine.communicate {
+            let tcx = &{this.tcx.tcx};
+
+            let buf = this.force_ptr(this.read_scalar(buf_op)?.not_undef()?)?;
+            let size = this.read_scalar(size_op)?.to_usize(&*this.tcx)?;
+            // If we cannot get the current directory, we return null
+            if let Ok(cwd) = env::current_dir() {
```

Comment:
Leave a FIXME that technically we have to set the `errno` global in case of errors

Reply:
That variable btw already exists: `last_error` in our `Evaluator`.

Reply:
oh sorry I didn't refreshed this